sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1058,7 +1068,17 @@ object CodeGenerator extends Logging {
 
     try {
       evaluator.cook("generated.java", code.body)
-      recordCompilationStats(evaluator)
+      val methodsToByteCodeSize = updateAndGetCompilationStats(evaluator)
+      methodsToByteCodeSize.foreach { case (name, byteCodeSize) =>
+        if (byteCodeSize > hugeMethodLimit) {
+          val clazzName = evaluator.getClazz.getSimpleName
+          val methodName = name.replace("$", "")
+          throw new IllegalArgumentException(
+            s"the size of $clazzName.$methodName is $byteCodeSize and this value goes over " +
+              s"the HugeMethodLimit $hugeMethodLimit (JVM doesn't compile methods " +
+              "larger than this limit)")
```

Comment:
We designed to show exception messages (see below the handling of `JaninoRuntimeException`), before falling back to non whole-stage codegen execution. I think we should follow the same behavior.

This exception will be hidden from user for now in `WholeStageCodegenExec`.

Reply:
ok